pkg/scheduler/apis/config/v1beta1/defaults.go

```diff
@@ -267,6 +267,13 @@ func SetDefaults_RequestedToCapacityRatioArgs(obj *v1beta1.RequestedToCapacityRa
 		// If no resources specified, used the default set.
 		obj.Resources = append(obj.Resources, defaultResourceSpec...)
 	}
+
+	// If resource weight is 0, use default weight(1) instead.
+	for i := range obj.Resources {
+		if obj.Resources[i].Weight == 0 {
+			obj.Resources[i].Weight = 1
+		}
+	}
```

Comment:
I doubt that the defaults here is necessary, before the defaults here got set, scheduler configuration has been validated here, 
https://github.com/kubernetes/kubernetes/blob/0abb908638dcea611253ed1dfc6d66e8b31d42d4/pkg/scheduler/apis/config/validation/validation_pluginargs.go#L261-L270

which means the weight cannot be zero at all if the resource is set in the configuration file, and the default resource has been set right above, which should be only `cpu` and `memory` with `weight` being set as `1`.
```obj.Resources = append(obj.Resources, defaultResourceSpec...)```

So, for what reason, the extra check here is needed?

Reply:
validation is done on the internal type after defaulting. This defaulting is necessary.

Reply:
can we have the resource other than the default resource for the internal type?

Reply:
sorry I am not sure I understand the question, what problem are you trying to solve?

Reply:
Never mind, I understand why are the values swapped here, set the weight to zero is not allowed without this, but now the default weight(1) is applied instead.

I think this should be documented somewhere as this is user-facing behavior change.

And why this is only done for `RequestedToCapacityRatio`, how about the `NodeResourcesLeastAllocated` and `NodeResourcesMostAllocated`?

Reply:
This is not a new behavior, it was originally done inside the plugin instantiation logic.

We are applying the same initialization in the new plugin for all scoring strategies.

Reply:
true, thanks for the explanation, let's mark this as resolved.